x-pack/plugins/observability_solution/infra/server/plugin.ts

```diff
@@ -268,7 +290,7 @@ export class InfraServerPlugin
       ]);
     }
 
-    registerRuleTypes(plugins.alerting, this.libs, this.config);
+    registerRuleTypes(plugins.alerting, this.libs, this.config, locators);
```

Comment:
**nit:** how about this? While typescript would complain in case `locator` doesn't meet the expected type definition, we could also just build the object directly here. up to you

```suggestion
    registerRuleTypes(plugins.alerting, this.libs, this.config, {
      alertsLocator,
      assetDetailsLocator,
      metricsExplorerLocator,
      inventoryLocator,
    });
```